### PR TITLE
i18n(zh-CN): Update translations in `v3.mdx`, `images.mdx` and `typescript.mdx`

### PR DESCRIPTION
**Pull request by Genteure**
#### What kind of changes does this PR include?

- Translated content

#### Description

Translate content added in #4446, #4447, #4495 and #4498.

### CODE REVIEW

File: src/content/docs/zh-cn/guides/upgrade-to/v3.mdx [hunk 1]
```diff
@@ -176,7 +176,17 @@ Astro v3.0 完全删除了这些已弃用的 API。它们不能再在你的 Astr
 - 废弃的配置选项：请参阅 [0.26 迁移指南](/zh-cn/guides/upgrade-to/v1/#new-configuration-api)
 - 废弃的脚本/样式类型：请参阅 [0.26 迁移指南](/zh-cn/guides/upgrade-to/v1/#new-default-script-behavior)
 
-### 移除： 内容集合 schema 中来自 `astro:content` 的 `image`
+### 移除：服务器代码中 Web API 的不完整垫片（shim）
```
Comment (Genteure) [inline src/content/docs/zh-cn/guides/upgrade-to/v3.mdx:179]: I don't like this translation for "shims".

This is a literal translation based on [the Wikipedia article "垫片 (程序设计)"](https://zh.wikipedia.org/wiki/%E5%9E%AB%E7%89%87_(%E7%A8%8B%E5%BA%8F%E8%AE%BE%E8%AE%A1)).

"Polyfill" is well understood and usually left untranslated, maybe "shim" should also be left as is.

**Comment by liruifengv** [comment]
建议同个文件合并到一个PR进行提交，一旦 PR被合并之后，该页面的翻译状态就会被标记成完成

File: src/content/docs/zh-cn/guides/upgrade-to/v3.mdx (hunk 1 above)
Comment (liruifengv) [reply src/content/docs/zh-cn/guides/upgrade-to/v3.mdx:179]: Yes, it's hard to translate.

We can do that for now.

**Review by liruifengv — APPROVED** [review]
LGTM

**Comment by liruifengv** [comment]
@Genteure 我在英文原文有一个新提交：https://github.com/withastro/docs/pull/4464

**Comment by Genteure** [comment]
I have updated this PR accordingly.

**Review by liruifengv — APPROVED** [review]

**Comment by Genteure** [comment]
I have rebased the changes onto current `main` and added changes made in those two PRs.

**Review by 100gle — APPROVED** [review]
LGTM!

**Review by Yan-Thomas — APPROVED** [review]
LGTM! Thanks peeps 🚀

**Pull request merged**